ckanext/datastore/backend/postgres.py

```diff
@@ -261,7 +261,6 @@ def _get_unique_key(context: Context, data_dict: dict[str, Any]) -> list[str]:
         AND a.attnum = ANY(idx.indkey)
         AND t.relkind = 'r'
         AND idx.indisunique = true
-        AND idx.indisprimary = false
```

Comment:
this function is also used in `create_indexes` so this change would break that functionality. Better to only modify `upsert_data` instead